From 930cf4117ee0a80a65510868ce978fd3d99a49c2 Mon Sep 17 00:00:00 2001
From: George G <kreach3r@users.noreply.github.com>
Date: Tue, 3 Nov 2015 23:57:22 +0200
Subject: [PATCH 3/9] Themes: Settings: Expose storage summary text

forward ported and adapted to oreo

This commit simply exposes the storage summary text hardcodes to allow
themers to theme them without issue.

nathanchance edit: updated for Nougat 7.1

Change-Id: I10c124b6a1516a1be7ac234c1dd7dbe2cbe05bb2
Signed-off-by: Adarsh-MR <adarshmr1998@gmail.com>
Signed-off-by: Harsh Shandilya <harsh@prjkt.io>
---
 res/layout/storage_summary.xml                           | 2 +-
 res/values/projekt_colors.xml                            | 4 ++++
 .../settings/deviceinfo/StorageSummaryPreference.java    | 9 +++++----
 .../settings/deviceinfo/StorageVolumePreference.java     | 7 ++++---
 4 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/res/layout/storage_summary.xml b/res/layout/storage_summary.xml
index d48d62d6ce..ca43ce4def 100644
--- a/res/layout/storage_summary.xml
+++ b/res/layout/storage_summary.xml
@@ -36,7 +36,7 @@
         android:singleLine="true"
         android:textAlignment="viewStart"
         android:textAppearance="@android:style/TextAppearance.Material.Subhead"
-        android:textColor="?android:attr/colorAccent"
+        android:textColor="@color/storage_summary_title_color"
         android:textSize="36sp"/>
 
     <TextView
diff --git a/res/values/projekt_colors.xml b/res/values/projekt_colors.xml
index 47c84a2665..10c7fd0fe6 100644
--- a/res/values/projekt_colors.xml
+++ b/res/values/projekt_colors.xml
@@ -22,4 +22,8 @@
     <color name="storage_private_3">#fff2a600</color>
     <color name="storage_private_4">#ffec407a</color>
     <color name="storage_private_5">#ffc0ca33</color>
+
+    <!-- Storage Summary Title -->
+    <color name="storage_summary_title_color">@*android:color/accent_device_default_light</color>
+
 </resources>
diff --git a/src/com/android/settings/deviceinfo/StorageSummaryPreference.java b/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
index 2cf98ee77a..4cb47a1f72 100644
--- a/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
+++ b/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
@@ -24,17 +24,20 @@ import android.util.MathUtils;
 import android.view.View;
 import android.widget.ProgressBar;
 import android.widget.TextView;
-
+import com.android.settings.Utils;
 import com.android.settings.R;
 
 public class StorageSummaryPreference extends Preference {
     private int mPercent = -1;
+    private int mSecondaryColor;
 
     public StorageSummaryPreference(Context context) {
         super(context);
 
         setLayoutResource(R.layout.storage_summary);
         setEnabled(false);
+
+        mSecondaryColor = Utils.getColorAttr(context, android.R.attr.textColorSecondary);
     }
 
     public void setPercent(long usedBytes, long totalBytes) {
@@ -54,9 +57,7 @@ public class StorageSummaryPreference extends Preference {
         }
 
         final TextView summary = (TextView) view.findViewById(android.R.id.summary);
-/*
-        summary.setTextColor(Color.parseColor("#8a000000"));
-*/
+        summary.setTextColor(mSecondaryColor);
 
         super.onBindViewHolder(view);
     }
diff --git a/src/com/android/settings/deviceinfo/StorageVolumePreference.java b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
index 657ae579c1..ab803835c2 100644
--- a/src/com/android/settings/deviceinfo/StorageVolumePreference.java
+++ b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
@@ -45,6 +45,7 @@ public class StorageVolumePreference extends Preference {
     private final VolumeInfo mVolume;
 
     private int mColor;
+    private int mSecondaryColor;
     private int mUsedPercent = -1;
 
     // TODO: ideally, VolumeInfo should have a total physical size.
@@ -55,6 +56,8 @@ public class StorageVolumePreference extends Preference {
         mVolume = volume;
         mColor = color;
 
+        mSecondaryColor = Utils.getColorAttr(context, android.R.attr.textColorSecondary);
+
         setLayoutResource(R.layout.storage_volume);
 
         setKey(volume.getId());
@@ -108,9 +111,7 @@ public class StorageVolumePreference extends Preference {
     public void onBindViewHolder(PreferenceViewHolder view) {
         final ImageView unmount = (ImageView) view.findViewById(R.id.unmount);
         if (unmount != null) {
-/*
-            unmount.setImageTintList(ColorStateList.valueOf(Color.parseColor("#8a000000")));
-*/
+            unmount.setImageTintList(ColorStateList.valueOf(mSecondaryColor));
             unmount.setOnClickListener(mUnmountListener);
         }
 
-- 
2.18.0

